### PR TITLE
SDK-266 [Node.js SDK] File storage README

### PR DESCRIPTION
**Pull request by RommeroInCountry**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-266](https://incountry.atlassian.net/browse/SDK-266)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version in package.json? What is the new version?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -631,8 +631,8 @@ const deleteResult = await storage.delete(countryCode, recordKey);
 
 ## Attaching files to a record
 
-General info about files, limit and how they would be stored if one decided to upload.
```
Comment (vadim-rutkevich-incountry) [inline README.md:634]: Please use:
- General information about files, their limits and how they would be stored if you decide to upload them.

File: README.md [hunk 2]
```diff
@@ -631,8 +631,8 @@ const deleteResult = await storage.delete(countryCode, recordKey);
...
-Attachment info will be available in record `StorageRecord` object `attachments` field.
+InCountry Storage allows to attach files to previously created records. Attachments meta information is available via `attachments` field of `StorageRecord` object.
```
Comment (vadim-rutkevich-incountry) [inline README.md:635]: Please use this:
InCountry Storage allows you to attach files to the previously created records. Attachments' meta information is available through the `attachments` field of the `StorageRecord` object.

File: README.md [hunk 3]
```diff
@@ -653,11 +653,10 @@ type StorageRecordAttachment = {
 
 
 ### Adding attachment
-Using `addAttachment` method you can add or replace attachment.
-Data can be provided as `Readable` stream, `Buffer` or via `string` with path to the file.
+`addAttachment` method allows you to add or replace attachments.
```
Comment (vadim-rutkevich-incountry) [inline README.md]: Please correct:
The `addAttachment` method allows you to add or replace attachments.

File: README.md [hunk 4]
```diff
@@ -653,11 +653,10 @@ type StorageRecordAttachment = {
...
+File data can be provided either as `Readable` stream, `Buffer` or `string` with path to the file in file system.
```
Comment (vadim-rutkevich-incountry) [inline README.md:658]: Please update:
File data can be provided either as `Readable` stream, `Buffer` or `string` with a path to the file in the file system.

File: README.md [hunk 5]
```diff
@@ -674,9 +673,15 @@ async addAttachment(
...
 ```
 
-### Deleting attachment
+Example of usage:
 ```typescript
+???
+```
 
+### Deleting attachment
+`deleteAttachment` method allows you to delete attachments using attachment's `fileId`.
```
Comment (vadim-rutkevich-incountry) [inline README.md]: Please update:
The `deleteAttachment` method allows you to delete attachments by using the attachment's `fileId`.

File: README.md [hunk 6]
```diff
@@ -687,8 +692,13 @@ deleteAttachment(
...
 
-### Downloading attachment content
-Using `getAttachmentFile` method you can download content of the attachment.
+Example of usage:
+```typescript
+???
+```
+
+### Downloading attachment
+`getAttachmentFile` method allows you to download attachment contents.
```
Comment (vadim-rutkevich-incountry) [inline README.md]: Please update:
The `getAttachmentFile` method allows you to download attachments.

File: README.md [hunk 7]
```diff
@@ -702,7 +712,13 @@ async getAttachmentFile(
 }
 ```
 
+Example of usage:
+```typescript
+???
+```
+
 ### Working with attachment meta info
+`getAttachmentMeta` method allows you can to retrieve attachment's metadata using attachment's `fileId`.
```
Comment (vadim-rutkevich-incountry) [inline README.md]: Please update:
The `getAttachmentMeta` method allows you to retrieve the attachment's metadata by using the attachment's `fileId`.

File: README.md [hunk 8]
```diff
@@ -714,7 +730,16 @@ async getAttachmentMeta(
 }
 ```
 
+Example of usage:
 ```typescript
+???
+```
+
+`updateAttachmentMeta` method allows you to update attachment's metadata (mimetype and filename).
```
Comment (vadim-rutkevich-incountry) [inline README.md]: Please update:
The `updateAttachmentMeta` method allows you to update the attachment's metadata (MIME type and file name).

File: README.md [hunk 9]
```diff
@@ -674,9 +673,15 @@ async addAttachment(
 }
 ```
 
-### Deleting attachment
+Example of usage:
 ```typescript
+???
+```
 
+### Deleting attachment
```
Comment (vadim-rutkevich-incountry) [inline README.md:691]: Please update:
### Deleting attachments

File: README.md [hunk 10]
```diff
@@ -8,30 +8,30 @@
...
 ----
-For a full list of supported countries and their codes please [follow this link](countries.md).
+To get the full list of supported countries and their codes, please [follow this link](countries.md).
 
 
 Usage
 -----
 
-To access your data in InCountry using NodeJS SDK, you need to create an instance of `Storage` class using async factory method `createStorage`.
+To access your data in InCountry Platform by using NodeJS SDK, you need to create an instance of the `Storage` class using the `createStorage` async factory method.
```
Comment (VladimirLevitskiy) [inline README.md:25]: Is the capital letter at the word 'Platform' correct?

File: README.md [hunk 11]
```diff
@@ -255,7 +255,7 @@ serviceKey1
 serviceKey2
 ```
 ##### String fields, hashed if Storage options "hashSearchKeys" is set to true (by default it is):
-**WARNING** If `hashSearchKeys` is set to `false` the following string fields will have length limitation of 256 chars at most
+**WARNING** If the `hashSearchKeys` option is set to `false` the following string fields will have length limitation of 256 characters at most.
```
Comment (VladimirLevitskiy) [inline README.md:258]: 256 or 200?

File: README.md [hunk 12]
```diff
@@ -462,22 +462,22 @@ The next exact match filtering criteria available:
...
 ```typescript
 // Matches all records where record.rangeKey >= 5 AND record.rangeKey <= 100
 { rangeKey1: { $gte: 5, $lte: 100 } }
 ```
 
 ##### Partial text match search
 
-Also you can search records by partial match using `searchKeys` operator, which performs partial match
-search (similar to `LIKE` SQL operator) among record's text fields `key1, key2, ..., key10`.
+You can also look up for data records by partial match using the `searchKeys` operator which performs partial match
```
Comment (VladimirLevitskiy) [inline README.md:473]: special symbols are not supported, I recommend to add a phrase about it

**Review by VladimirLevitskiy — CHANGES_REQUESTED** [review]

File: README.md (hunk 10 above)
Comment (RommeroInCountry) [reply README.md:25]: yep, this one is intentional

File: README.md (hunk 11 above)
Comment (RommeroInCountry) [reply README.md:258]: 256. 200 is for `search_keys` value

**Review by VladimirLevitskiy — APPROVED** [review]
LGTM

File: README.md [hunk 13]
```diff
@@ -119,14 +119,14 @@ const storage = new Storage({
...
+The `validate` method fetches the secret using `GetSecretsCallback` and validates it. If custom encryption configurations were provided they would also be checked with all the matching secrets.
 
 
 #### oAuth Authentication
 
-SDK also supports oAuth authentication credentials instead of plain API key authorization. oAuth authentication flow is mutually exclusive with API key authentication - you will need to provide either API key or oAuth credentials.
+The SDK also supports oAuth authentication credentials instead of plain API key authorization. oAuth authentication flow is mutually exclusive with API key authentication - you will need to provide either API key or oAuth credentials.
 
-Below is the example how to create storage instance with oAuth credentials (and also provide custom oAuth endpoint):
+Below you can find the example of how to  create a storage instance with oAuth credentials (and also provide a custom oAuth endpoint):
```
Comment (artemgolozubov) [inline README.md:129]:
```suggestion
Below you can find the example of how to create a storage instance with oAuth credentials (and also provide a custom oAuth endpoint):
```

File: README.md [hunk 14]
```diff
@@ -462,22 +462,22 @@ The next exact match filtering criteria available:
...
 // Matches all records where record.rangeKey >= 5 AND record.rangeKey <= 100
 { rangeKey1: { $gte: 5, $lte: 100 } }
 ```
 
 ##### Partial text match search
 
-Also you can search records by partial match using `searchKeys` operator, which performs partial match
-search (similar to `LIKE` SQL operator) among record's text fields `key1, key2, ..., key10`.
+You can also look up for data records by partial match using the `searchKeys` operator which performs partial match
+search (similar to the `LIKE` SQL operator, without special symbols) within records text fields `key1, key2, ..., key10`.
```
Comment (artemgolozubov) [inline README.md:474]:
```suggestion
search (similar to the `LIKE` SQL operator, without special characters) within records text fields `key1, key2, ..., key10`.
```

File: README.md [hunk 15]
```diff
@@ -674,9 +675,22 @@ async addAttachment(
 }
 ```
 
-### Deleting attachment
+Example of usage:
 ```typescript
+// with file path
+await storage.addAttachment(COUNTRY, recordData.recordKey, { file: '../file' });
+
+//with data in Stream
```
Comment (artemgolozubov) [inline README.md:683]:
```suggestion
// using data in Stream
```

File: README.md [hunk 16]
```diff
@@ -674,9 +675,22 @@ async addAttachment(
 }
 ```
 
-### Deleting attachment
+Example of usage:
 ```typescript
+// with file path
```
Comment (artemgolozubov) [inline README.md:680]:
```suggestion
// using file path
```

File: README.md [hunk 17]
```diff
@@ -687,22 +701,39 @@ deleteAttachment(
...
+
+const { attachmentData } = await storage.getAttachmentFile(COUNTRY, recordData.recordKey, attachmentMeta.fileId);
+
+const { file, fileName } = attachmentData;
+const writeStream = fs.createWriteStream(`./${fileName}`);
+file.pipe(writeStream);
+```
+
 ### Working with attachment meta info
+The `getAttachmentMeta` method allows you can to retrieve attachment's metadata using attachment's `fileId`.
```
Comment (artemgolozubov) [inline README.md:736]: Fix this phrase pls

**Review by artemgolozubov — CHANGES_REQUESTED** [review]
Pls fix the phrase describing getAttachmentMeta

**Review by VladimirLevitskiy — APPROVED** [review]

**Review by artemgolozubov — APPROVED** [review]
LGTM

**Pull request merged**